src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureKeyVaultReference/AzureKeyVaultSecretProvider.cs

```diff
@@ -115,8 +115,13 @@ private SecretClient GetSecretClient(Uri secretUri)
                 return null;
             }
 
-            client = new SecretClient(new Uri(secretUri.GetLeftPart(UriPartial.Authority)), _keyVaultOptions.Credential);
+            client = new SecretClient(
+                new Uri(secretUri.GetLeftPart(UriPartial.Authority)),
+                _keyVaultOptions.Credential,
+                _keyVaultOptions.ClientOptions ?? new SecretClientOptions());
```

Comment:
The `SecretClient` class seems to expect null to be passed for `options`, so we don't need the null check here necessarily. I don't mind it this way, but you could simplify it if you want.

Reply:
> The SecretClient class seems to expect null to be passed for options

Yea, I know that. 
The reason I add null check here is that I want to make it more robust and don't rely on KeyVault lib to handle the null. 
Besides, if user reads our code, they will know the default option will be used without going to the code base of keyvault lib.